### PR TITLE
feat: reuse pydantic v2 TypeAdapter instance

### PR DESCRIPTION
**Pull request by unights**
Solves issue #281

### CODE REVIEW

File: taskiq/compat.py [hunk 1]
```diff
@@ -13,8 +14,12 @@
 if IS_PYDANTIC2:
     T = TypeVar("T")
 
+    @lru_cache(maxsize=None)
```
Comment (vvanglro) [inline taskiq/compat.py:17]: Don't set it to None, which in long-running task libraries means that the cache will grow indefinitely.

File: taskiq/compat.py [hunk 2]
```diff
@@ -13,8 +14,12 @@
...
+    def create_type_adapter(annot: T) -> pydantic.TypeAdapter:
```
Comment (s3rius) [inline taskiq/compat.py:18]: Should be

```suggestion
    def create_type_adapter(annot: T) -> pydantic.TypeAdapter[T]:
```

**Review by s3rius — CHANGES_REQUESTED** [review]

File: taskiq/compat.py (hunk 2 above)
Comment (unights) [reply taskiq/compat.py]: Oh yeah, I forgot it. Thanks for pointing out it.

File: taskiq/compat.py (hunk 1 above)
Comment (unights) [reply taskiq/compat.py:17]: I thought the number of parameter types is usually limited. But this is indeed a dangerous action. Should we set it to the default 128, or a large value?

Comment (vvanglro) [reply taskiq/compat.py:17]: I suggest default.

**Review by s3rius — APPROVED** [review]

**Comment by s3rius** [comment]
Thank you very much for your contribution.

**Pull request merged**